identifier.go

```diff
@@ -80,24 +79,24 @@ func encodeIdentifier(identifier Identifier) []byte {
 
 		tag := identifier.Tag
 
-		highBit := uint(63)
-		for {
-			if tag&(1<<highBit) != 0 {
-				break
-			}
-			highBit--
-		}
+		b = append(b, encodeHighTag(tag)...)
+	}
+	return b
+}
 
-		tagBytes := int(math.Ceil(float64(highBit) / 7.0))
-		for i := tagBytes - 1; i >= 0; i-- {
-			offset := uint(i) * 7
-			mask := Tag(0x7f) << offset
-			tagByte := (tag & mask) >> offset
-			if i != 0 {
-				tagByte |= 0x80
-			}
-			b = append(b, byte(tagByte))
+func encodeHighTag(tag Tag) []byte {
+	b := make([]byte, 0, 4)
+	for tag != 0 {
+		t := tag & HighTagValueBitmask
+		tag >>= 7
+		if len(b) != 0 {
+			t |= HighTagContinueBitmask
 		}
+		b = append(b, byte(t))
+	}
+	// reverse
```

Comment:
... and I have no idea why we have to reverse it here, (endianness?) but when I take the time to actually understand this I assume it will make sense.